agent/src/main/java/kanela/agent/InstrumentationLoader.java

```diff
@@ -51,10 +55,14 @@
         });
     }
 
-    private static Option<KanelaInstrumentation> loadInstrumentation(String instrumentationClassName, ClassLoader classLoader) {
+    private static Option<KanelaInstrumentation> loadInstrumentation(KanelaConfiguration configuration, KanelaConfiguration.ModuleConfiguration moduleConfiguration, String instrumentationClassName, ClassLoader classLoader) {
         Logger.info(() -> format(" ==> Loading {0} ", instrumentationClassName));
         try {
-            return Option.some((KanelaInstrumentation) Class.forName(instrumentationClassName, true, classLoader).newInstance());
+            Class<?> instrumentationClass = Class.forName(instrumentationClassName, true, classLoader);
+            if (configuration.getInstrumentationRegistryConfig().isEnabled()) {
+                InstrumentationRegistryListener.instance().registerModuleVersion(moduleConfiguration.getKey(), Manifests.getJarVersion(instrumentationClass));
```

Comment:
Also, the name `InstrumentationRegistryListener` is really a `Listener` or have other responsabilities, maybe only `InstrumentationRegistry` is a better name?

Reply:
+1